internal/catalogmetadata/client/client.go

```diff
@@ -12,15 +13,46 @@ import (
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 )
 
-func NewClient(cl client.Client) *Client {
-	return &Client{cl: cl}
+// TODO: When a catalogd release containing https://github.com/operator-framework/catalogd/pull/168
+// is made this can be removed in favor of parsing `Catalog.Status` for the
+// appropriate URL for fetching catalog contents.
+const catalogdOnClusterBaseURL = "http://catalogd-catalogserver.catalogd-system.svc"
+const catalogdOnClusterURLTemplate = "%s/catalogs/%s/all.json"
```

Comment:
It's probably worth waiting for that to go in, and to release a v0.6.1 of catalogd for using here

Reply:
I'm fine with waiting if that's what we really want to hold this on that. 

It looks like the next catalogs release will have to be v0.7.0 since a release branch wasn't created for the v0.6.0 release and a change warranting a minor increment just merged to main. Looks like catalogd is also missing a release guide

Reply:
> release a v0.6.1 of catalogd for using here

That change should _probably_ be v0.7.0 since we're adding a new feature/API field. It's generally a really good habit to only put actual bug fixes in patch releases.

> a release branch wasn't created for the v0.6.0

We can always go back and branch off from the `v0.6.0` commit if we need to patch that minor release.

Reply:
To @anik120's point though. This does feel like the wrong abstraction knowing that we're planning to have the URL provided by the individual catalog objects.

It looks like this setup is _just_ to facilitate test code, right?

An alternative could be an `http.RoundTripper` wrapper that replaces the request base URL with the test server's URL. That way test-specific needs don't leak into the Client API.

Reply:
>It looks like this setup is just to facilitate test code, right?

No. This setup is specifically because the version of catalogd this is based on doesn't have a status field to show the URL to hit. While that is definitely nice to have, it's not absolutely necessary if you know the pattern for building the correct URL for this version of catalogd. My intention here was to go ahead and get the use of the HTTP server in place by setting these constants that will be removed and replaced when a version of catalogd is used that gives the URL in a Catalog's status.

Reply:
I took a look at the `http.RoundTripper` and it seems like that would also be leaking test-specific needs into the Client API but I'm happy to use that if it is preferred over the current approach. That being said, it seems like all of this would be moot once Anik's PR in catalogd is merged to add the status field for the URL so if we are going to hold this PR on that I'd prefer we don't do any further work in this specific area of the Client API

Reply:
What is the specific goal of having this repo (in an unreleased `main` state) depend on versioned/released commits from `catalogd`? What is lost if both repos float in an unreleased state between releases?

Reply:
>What is the specific goal of having this repo (in an unreleased main state) depend on versioned/released commits from catalogd? What is lost if both repos float in an unreleased state between releases?

No specific goal, just the way this project has been configured. As far as I am aware, all the testing infrastructure and scripts for installing and running operator-controller specifically look for releases from it's dependencies. More explicitly, this Makefile target: https://github.com/operator-framework/operator-controller/blob/6640aacac00f183d861a64d148f4636df1ffb964/Makefile#L139-L142

calls this shell script that installs manifests from a released version of catalogd, rukpak, and cert-manager: https://github.com/operator-framework/operator-controller/blob/6640aacac00f183d861a64d148f4636df1ffb964/scripts/install.tpl.sh#L39-L48

I don't see why we couldn't update everything to make it so operator-controller can depend on unversioned/unreleased commits from it's dependencies but that is significantly out of scope for this PR and, IMO, waiting for a new release is going to be the quickest path to pick up these changes.